exporter/instanaexporter/internal/otlptext/databuffer.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlptext // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/otlptext"
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type dataBuffer struct {
```

Comment:
This code seemed familiar, and I found it here:

https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/loggingexporter/internal/otlptext/databuffer.go

I would either suggest splitting the original otlptext package into a public utility package or at least leaving a comment pointing people to the original resource so that any fixes (especially security) could be ported to this version.

Reply:
Thanks for catching this. As this code isn't originally mine, I hadn't realised that it was copied. 

For the moment, I have added a comment to the top of the file specifying its a copy and to keep uptodate. In medium to long term, we need to either refactor the code to avoid the duplication oro work to expose the collector code. Whichever makes most sense.